brd-android/app/src/main/java/com/breadwallet/tools/util/LinkExtensions.kt

```diff
@@ -80,11 +81,17 @@ suspend fun String.asLink(
             .firstOrNull() ?: (null to null)
 
     return when {
-        address != null && currencyCode != null && currencyCode != "xlm" ->
+        address != null && currencyCode != null && currencyCode != "xlm" -> {
+            val amount = if (uri.queryParameterNames.contains("amount")) {
+                BigDecimal(uri.getQueryParameter("amount"))
+            } else null
+
             Link.CryptoRequestUrl(
                 currencyCode = currencyCode,
-                address = address.toString()
+                address = address.toString(),
+                amount = amount
```

Comment:
Could be simplified:

```suggestion
                amount = uri.getQueryParameter("amount")?.let { BigDecimal(it) }
```